### PR TITLE
[FLINK-11356][tests] Port JobManagerStartupTest to new code base

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

- Moved JobManagerStartupTest#testStartupWithPortInUse to
BootstrapToolsTest#testActorSystemInstantiationFailureWhenPortOccupied

- Moved JobManagerStartupTest#testJobManagerStartupFails to
BlobServerTest#testFailureIfStorageDirectoryCannotBeCreated

## Verifying this change

- Executed the affected test cases

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/BootstrapToolsTest.java [hunk 1]
```diff
@@ -360,4 +368,32 @@ public void testConcurrentActorSystemCreation() throws Exception {
...
+	public void testActorSystemInstantiationFailureWhenPortOccupied() throws Exception {
+		ServerSocket portOccupier;
+		final int port;
+
+		try {
+			port = NetUtils.getAvailablePort();
+			portOccupier = new ServerSocket(port, 10, InetAddress.getByName("0.0.0.0"));
+		}
+		catch (Throwable t) {
+			// could not find free port, or open a connection there
```
Comment (GJL) [inline flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/BootstrapToolsTest.java]: It's acceptable to silently fail if we cannot open a socket?

File: flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobServerTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	@ClassRule
+	public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+	/**
+	 * Tests that the {@link BlobServer} fails if the blob storage directory
+	 * cannot be created.
+	 */
+	@Test
+	public void testFailureIfStorageDirectoryCannotBeCreated() throws IOException {
+		assumeTrue(!OperatingSystem.isWindows()); //setWritable doesn't work on Windows.
```
Comment (GJL) [inline flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobServerTest.java]: I would prefer to move the _assumption_ closer to `createNonWritableDirectory`. This makes the comment clearer. Also consider `assumeFalse` as `!` can be overlooked.
```suggestion
		assumeFalse(OperatingSystem.isWindows()); //setWritable doesn't work on Windows.
		final File blobStorageDirectory = createNonWritableDirectory();

		final Configuration configuration = new Configuration();
		final String nonExistDirectory = new File(blobStorageDirectory, "does_not_exist_for_sure").getAbsolutePath();
		configuration.setString(BlobServerOptions.STORAGE_DIRECTORY, nonExistDirectory);
```

File: flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/BootstrapToolsTest.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/BootstrapToolsTest.java]: I think you're right that this not necessary by not preselecting the `port`. Instead we should let the `ServerSocket` pick a port via `ServerSocket(0,...)`. Will change it accordingly.

File: flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobServerTest.java (hunk 2 above)
Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobServerTest.java]: Good point Gary. I will move the assumption into the `createNonWritableDirectory` method and change it to `assumeFalse`.

**Comment by tillrohrmann** [comment]
Thanks for the review @GJL. I've addressed your comments.

**Pull request closed without merging**